### PR TITLE
feat: Add quickstart/postgres to gh actions

### PR DESCRIPTION
**Pull request by bradbeam**
Signed-off-by: Brad Beam <brad.beam@carbonrelay.com>

### CODE REVIEW

File: .github/workflows/push.yaml [hunk 1]
```diff
@@ -53,3 +53,43 @@ jobs:
...
+      - name: KinD (Kubernetes in Docker) Action
+        uses: engineerd/setup-kind@v0.3.0
+      - name: commands
+        run: |
+          kubectl get nodes -o wide
+          kubectl get node kind-control-plane -o yaml
+          kubectl get pods -n kube-system
+      - name: redsky quickstart
+        run: |
+          echo "Install redskyctl"
```
Comment (bradbeam) [inline .github/workflows/push.yaml]: We should probably skip this and use the artifact we just built

**Comment by bradbeam** [comment]
This adds ~4-5min to the build but I think it gives a good sanity check that things work as expected.

File: .github/workflows/push.yaml (hunk 1 above)
Comment (jgustie) [reply .github/workflows/push.yaml]: I think you can use:

```
- uses: actions/download-artifact@v2
  with:
    name: redskyctl_linux_amd64
    path: ...
```

File: .github/workflows/push.yaml [hunk 2]
```diff
@@ -53,3 +53,43 @@ jobs:
...
+          curl -sLo - https://github.com/redskyops/redskyops-controller/releases/download/v1.5.0/redskyctl-linux-amd64.tar.gz | \
+            tar -zxf - && \
+            chmod 755 ./redskyctl
```
Comment (jgustie) [inline .github/workflows/push.yaml]: Was this necessary? The permissions should be preserved.

File: .github/workflows/push.yaml [hunk 3]
```diff
@@ -53,3 +53,43 @@ jobs:
...
+          echo "Init redskyops"
+          ./redskyctl init
+          echo "Install kustomize"
+          curl -s "https://raw.githubusercontent.com/\
+          kubernetes-sigs/kustomize/master/hack/install_kustomize.sh"  | \
+            bash
```
Comment (jgustie) [inline .github/workflows/push.yaml]: I wonder if the `kubectl -k` option is enough: we don't actually do much with these examples.

**Comment by jgustie** [comment]
4-5 minutes on top of the 8 that we have is getting pretty steep. Does it make sense to put this in a separate workflow?

File: .github/workflows/push.yaml (hunk 3 above)
Comment (bradbeam) [reply .github/workflows/push.yaml]: No go with `kubectl -k` unfortunately. Looks like Kubectl bundles kustomize v2 and we need v3 to work with our kustomize files. :(

File: .github/workflows/push.yaml (hunk 2 above)
Comment (bradbeam) [reply .github/workflows/push.yaml]: Should be able to drop this and use the precompiled version ( under `dist` ).

File: .github/workflows/push.yaml [hunk 4]
```diff
@@ -53,3 +58,32 @@ jobs:
...
+          echo "Init redskyops"
+          dist/redskyctl_linux_amd64/redskyctl init
+          echo "Install kustomize"
+          curl -s "https://raw.githubusercontent.com/\
+          kubernetes-sigs/kustomize/master/hack/install_kustomize.sh"  | \
+            bash
+          echo "Patch postgres deployment"
+          ./kustomize build github.com/redskyops/redskyops-recipes/postgres > postgres.yaml
+          # Need to change strategy here since we're limited to 2cpu total
+          sed -i -e '37i\  strategy:\n    type: Recreate' postgres.yaml
```
Comment (jgustie) [inline .github/workflows/push.yaml]: My vote here would be to create `/test/quickstart/kustomization.yaml` file:

```yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization

resources:
- github.com/redskyops/redskyops-recipes/postgres

patchesStrategicMerge:
- |-
  apiVersion: apps/v1
  kind: Deployment
  metadata:
    name: postgres
  spec:
    strategy:
      type: Recreate
```
That way you can just `kustomize build test/quickstart  | kubectl apply -f -`.

I'm on the fence about adding a `/test/quickstart/run.sh` to do everything from here down.

**Review by jgustie — APPROVED** [review]

**Pull request merged**